torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -506,83 +503,84 @@ def linear_transpose(
 
 
 def permute_linear_fusion(module: torch.fx.GraphModule) -> torch.fx.GraphModule:
-    for node in module.graph.nodes:
-        if node.op == "call_function" and node.target == torch.nn.functional.linear:
-            if len(node.args) > 0:
-                input_node = node.args[0]
+    for node in module.graph.find_nodes(
+        op="call_function", target=torch.nn.functional.linear
+    ):
+        if len(node.args) > 0:
+            input_node = node.args[0]
+        else:
+            input_node = node.kwargs["input"]
+        if (
+            input_node.op == "call_method"
+            and input_node.target == "permute"
+            and check_permute(input_node)
+        ):
+            normalized = NormalizedLinearNode(node)
+            if len(input_node.args) > 0:
+                input = input_node.args[0]
             else:
-                input_node = node.kwargs["input"]
-            if (
-                input_node.op == "call_method"
-                and input_node.target == "permute"
-                and check_permute(input_node)
-            ):
-                normalized = NormalizedLinearNode(node)
-                if len(input_node.args) > 0:
-                    input = input_node.args[0]
-                else:
-                    input = input_node.kwargs["input"]
-                weight = normalized.get_weight()
-                bias = normalized.get_bias()
-                with module.graph.inserting_before(node):
-                    fused_node = module.graph.call_function(
-                        transpose_linear, args=(input, weight, bias)
-                    )
-                    node.replace_all_uses_with(fused_node)
-                    module.graph.erase_node(node)
-                    if len(input_node.users) == 0:
-                        module.graph.erase_node(input_node)
+                input = input_node.kwargs["input"]
+            weight = normalized.get_weight()
+            bias = normalized.get_bias()
+            with module.graph.inserting_before(node):
+                fused_node = module.graph.call_function(
+                    transpose_linear, args=(input, weight, bias)
+                )
+                node.replace_all_uses_with(fused_node)
+                module.graph.erase_node(node)
+                if len(input_node.users) == 0:
+                    module.graph.erase_node(input_node)
 
     module.graph.lint()
     module.recompile()
     return module
 
 
 def permute_matmul_fusion(module: torch.fx.GraphModule) -> torch.fx.GraphModule:
-    for node in module.graph.nodes:
-        if node.op == "call_function" and (
-            node.target == torch.bmm or node.target == torch.matmul
+    for node in itertools.chain(
+        module.graph.find_nodes(op="call_function", target=torch.bmm),
+        module.graph.find_nodes(op="call_function", target=torch.matmul),
+    ):
```

Comment:
Would it be possible to merge the results from multiple queries so they are traversed in the original graph order? Otherwise by changing the order nodes are visited, we may in general change the result of an fx pass.

Reply:
@jansel and I were chatting about these passes should not require to be combined and could be applied one after each other. For the sake of consistency I can combine them, but having them one after each other is better for isolation?

Reply:
I agree that most passes should work in any order, but that doesn't mean that order isn't important. Say for example we have two patterns `x * 0 -> 0` and `x + 0 -> x` then apply these to the expression `x + (x * 0)`; if we visit the `+` op before the `*` then the add pattern doesn't match and we get `x + 0` vs. the other way around both match and we get simply `x`.